esdm/esdm_drng_mgr.c

```diff
@@ -618,9 +622,9 @@ static bool esdm_drng_check_disable_threshold(struct esdm_drng *drng)
 {
 	bool request_limit_reached = atomic_read_u32(&drng->requests_since_fully_seeded) >
 	    			     esdm_config_drng_max_wo_reseed();
-	bool bit_limit_reached = (esdm_config_drng_max_wo_reseed_bits() > 0) &&
-				 (atomic_read_u32(&drng->internal_bits) >
-				 ((uint32_t) esdm_config_drng_max_wo_reseed_bits()));
+	bool bit_limit_reached = (esdm_config_drng_max_wo_reseed_bits() != UINT32_MAX) &&
```

Comment:
What about simply removing the check for UINT32_MAX?

Reply:
Good point, as the number of bits can't get bigger than this and the condition will always be false. :+1: